### PR TITLE
Docs: deep dive explanation about sids

### PR DESCRIPTION
**Pull request by AlexandrHoroshih**

### CODE REVIEW

File: beta/src/content/docs/en/explanation/sids.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+The SIDs have to important properites:
+
+1. They are **unique** - each sid for each unit should be unique
+2. They are **stable** between different environments - each sid of each unit in some environment (e.g. server code) should be equal to a sid of this unit in any other environment (e.g. client code)
+
+## How to add SIDs {#how-to-add-sids}
+
+The SIDs can be added both manually and automatically, but it is important, that SIDs are set before any bundling happened - otherwise there is no way to guarantee stability.
+
+**Manual way**
```
Comment (sergeysova) [inline beta/src/content/docs/en/explanation/sids.md:21]:
```suggestion
### Manual way {#how-to-add-sids-manual}
```

File: beta/src/content/docs/en/explanation/sids.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+```tsx
+import { createStore } from "effector";
+
+export const $myStore = createStore(42, { sid: "my-stable-id" });
+```
+
+It is important, that all sids are also **unique**. If you are usign the manual way, you have to guarantee that yourself.
+
+**Automatic way**
```
Comment (sergeysova) [inline beta/src/content/docs/en/explanation/sids.md:31]:
```suggestion
### Automatic way {#how-to-add-sids-automatic}
```

File: beta/src/content/docs/en/explanation/sids.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+Thankfully, there are [`effector/babel-plugin`](/api/effector/babel-plugin) and [`@effector/swc-plugin`](https://github.com/effector/swc-plugin), which will provide SIDs automatically.
+
+Because code-transpilation tools are working at the file-level and are run before bundling happens - it is possible to make SIDs **stable** for every environment.
+
+:::tip
+It is preferable to use [`effector/babel-plugin`](/api/effector/babel-plugin) or [`@effector/swc-plugin`](https://github.com/effector/swc-plugin) instead of adding sids manually.
+:::
+
+## Why we need Stable Identifiers {#why-we-need-sids}
```
Comment (sergeysova) [inline beta/src/content/docs/en/explanation/sids.md:43]:
```suggestion
## Why do we need Stable Identifiers {#why-do-we-need-sids}
```

File: beta/src/content/docs/en/explanation/sids.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+And then we need to unpack these values at the client and add this "server-calucalted" state to a client-side instances of all these stores!
+
+#### Solution {#serialization-boundary-solution}
+
+This is a hard problem and to solve this, `effector` needs a way to connect "server-calculated" state of some store with its client-side instance.
+
+This where SIDs are required. Because SID is the same for the same store in any environment, `effector` can easily connect server-side values with client-side stores.
+
+**Example**
```
Comment (sergeysova) [inline beta/src/content/docs/en/explanation/sids.md:99]:
```suggestion
#### Example {#serialization-boundary-example}
```

File: beta/src/content/docs/en/explanation/sids.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+
+This is a generic server side rendering handler. The `renderHtmlString` function is an implementation detail, which will depend on the framework you use.
```
Comment (sergeysova) [inline beta/src/content/docs/en/explanation/sids.md:101]:
```suggestion
This is a generic server side rendering handler. The `renderHtmlToString` function is an implementation detail, which will depend on the framework you use.
```

File: beta/src/content/docs/en/explanation/sids.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+      lastName: "Doe",
+    },
+  });
+
+  // extract scope values to simple js object of `{ [storeSid]: storeState }`
+  const values = serialize(scope);
+
+  const serializedState = JSON.stringify(values);
+
+  return renderHtmlString({
```
Comment (sergeysova) [inline beta/src/content/docs/en/explanation/sids.md:126]:
```suggestion
  return renderHtmlToString({
```

File: beta/src/content/docs/en/explanation/sids.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+description: Stable IDentifiers
+redirectFrom:
+  - /en/explanation/sids
+  - /explanation/sids
+  - /docs/explanation/sids
+---
+
+The SID is a Stable IDentifier of effector unit. It can be used for different purposes, but the main use-case is a Server Side Rendering.
+
+The SIDs have to important properites:
```
Comment (victordidenko) [inline beta/src/content/docs/en/explanation/sids.md:12]:
```suggestion
The SIDs have two important properties:
```
or
```suggestion
The SIDs must have two important properties:
```
it is not clear, what meaning you wanted to express...

**Review by sergeysova — CHANGES_REQUESTED** [review]

**Pull request merged**